cumulus_library/cli.py

```diff
@@ -19,13 +19,13 @@
     errors,
     protected_table_builder,
     study_parser,
-    upload,
 )
+from cumulus_library.actions import builder, cleaner, exporter, file_generator, uploader
 from cumulus_library.template_sql import base_templates
 
 
 class StudyRunner:
-    """Class for managing cursors and executing Cumulus queries"""
+    """Class for managing cursors and invoking actions"""
```

Comment:
nit: I think there's opportunity here for even further isolation of the actions - like this file still holds a lot of info about what each action needs, which feels like it's breaking the abstraction a bit. In theory, the top level could just be "routing to a single action method which does the further CLI parsing" and "shared utility methods for CLI parsing". But that's not an important refactor. This is already quite a bit better.

Reply:
yeah, i don't disagree - i drew a line in the sand here for a couple reasons:
- i like two medium changes over one big one for something of this scope/keeping context in my brain
- the `lib_transactions` stuff currently is kind of inline here in a way that doesn't decouple well - there probably should be a dedicated module for managing that

But i generally think this is a good thing to do at some point in the future.